### ISSUE TITLE
BUG: pandas.Series.describe() displays confusing information

### ISSUE THREAD

**Issue by Iqigai** [pandas-dev/pandas#47580]
### Pandas version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pandas.pydata.org/docs/whatsnew/index.html) of pandas.

- [X] I have confirmed this bug exists on the main branch of pandas.

### Reproducible Example

```python
import pandas as pd
s = pd.Series([1, 2 ,3 ,4])
s.describe()
>>>count    4.000000
mean     2.500000
std      1.290994
min      1.000000
25%      1.750000
50%      2.500000
75%      3.250000
max      4.000000
dtype: float64
```

### Issue Description

The results displayed are a bit incoherent in their format and can create confusion.
First, the count is displayed as a float, although it is clearly an int. Same goes for the max and min values
Second, the dtype information which is displayed seems to correspond to the series returned by the describe() method but that creates confusion as what is expected is information on the Series itself, not meta data.
The display format are minor issues, but the dtype one can really create confusion as it can be thought of being the dtype of the Series calling the method.

### Expected Behavior

int values should be displayed as int
dtype should refer to the calling Series not the resulting one, or not being displayed at all.

### Installed Versions

<details>
INSTALLED VERSIONS
------------------
commit           : e8093ba372f9adfe79439d90fe74b0b5b6dea9d6
python           : 3.10.5.final.0
python-bits      : 64
OS               : Windows
OS-release       : 10
Version          : 10.0.19043
machine          : AMD64
processor        : Intel64 Family 6 Model 158 Stepping 10, GenuineIntel
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : English_Canada.1252
pandas           : 1.4.3
numpy            : 1.22.4
pytz             : 2022.1
dateutil         : 2.8.2
setuptools       : 62.4.0
pip              : 22.1.2
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : 8.4.0
pandas_datareader: None
bs4              : None
bottleneck       : None
brotli           : None
fastparquet      : None
fsspec           : None
gcsfs            : None
markupsafe       : None
matplotlib       : 3.5.2
numba            : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pyreadstat       : None
pyxlsb           : None
s3fs             : None
scipy            : 1.8.1
snappy           : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
zstandard        : None
</details>

**Comment by phofl** [comment]
Hi, 

thanks for your report. As the documentation states, the return value of describe is a Series, hence the float representation and the accompanying dtype.

**Comment by Iqigai** [comment]
Yes indeed, it returns a Series or DataFrame but the name suggests it is primarily descriptive statistics related to the input object. Ideally, it should display data that describe the input, not the output itself, but fair enough.
